src/core/wallet/services/JWTService/JWTProofService.ts

```diff
@@ -0,0 +1,57 @@
+import appendQuery from 'append-query';
+import { injectable } from 'tsyringe';
+import { ContractService } from '../contractService/contractsService';
+import { WalletService } from '../walletService/walletService';
+import { JWTService } from './JWTService';
+
+@injectable()
+export class JWTProofService {
+  constructor (private jwtService: JWTService, private walletService: WalletService, private contractService: ContractService) {
+
+  }
+
+  /**
+   * Create a certificate JWTProof
+   * @param certificateId
+   */
+  public createCertificateJWTProof =(certificateId: number) => {
+    return this.jwtService.sign({
+      sub: 'certificate',
+      subId: certificateId
+    });
+  }
+
+  /**
+   * Create JWTProof and add it to url
+   * @param url
+   * @param certificateId
+   */
+    public createActionJWTProofLink= (url:string, certificateId: number) => {
+      const token = this.createCertificateJWTProof(certificateId);
+
+      return appendQuery(url, { token });
+    }
+
+  /**
+   * Decode proof and return it
+   * @param token
+   */
+  public decodeJWTProof = (token) => {
+    return this.jwtService.decode(token);
+  };
+
+  /**
+   * Method to check if token is valid and if certificateId is own by current wallet
+   * @param token
+   */
+  public isCertificateJWTProofValid = async (token) => {
```

Comment:
Why do we need a function that check if the proof is valid for our wallet ?
We should check if the proof is valid with the certificate owner, no ?

Reply:
I ll add address as parameter